TailLight/device.cpp

```diff
@@ -130,28 +395,129 @@ Routine Description:
 --*/
 {
     UNREFERENCED_PARAMETER(OutputBufferLength);
-    UNREFERENCED_PARAMETER(InputBufferLength);
-
-    //KdPrint(("TailLight: EvtIoDeviceControl (IoControlCode=0x%x, InputBufferLength=%Iu)\n", IoControlCode, InputBufferLength));
 
+    RequestFlags incomingRequestFlags = (RequestFlags)WdfRequestGetInformation(Request);
```

Comment:
What kind of information are you trying to get in `incomingRequestFlags`?

I don't see WdfRequestSetInformation or WdfRequestCompleteWithInformation being called anywhere  here to populate the "information" parameter in advance.